#!/usr/bin/env python2
from dynamic_reconfigure.parameter_generator_catkin import *
PACKAGE = "asr_ism_visualizations"


gen = ParameterGenerator()

gen.add("markerLifetime", double_t, 0, "Time Markers are shown (0=permanent).", 0, 0, 120)
gen.add("baseFrame", str_t, 0, "Base Frame of scene for result visualization.", "/map")
gen.add("lineScale", double_t, 0, "Width of lines for recognitionresult visualization.", 0.02, 0, 0.1)
gen.add("treeDepthSize", double_t, 0, "Depth of one tree hierarchie for recognitionresult visualization.", 0.2, 0, 0.5)
gen.add("ismMarkerScale", double_t, 0, "Scaling of one ISM-Marker for recognitionresult visualization.", 0.05, 0, 0.2)
gen.add("isPoseRelative", bool_t, 0, "ISM-Tree is relative orienteted to pose from reference for recognitionresult visualization.", False)
gen.add("ignoreZOffset", bool_t, 0, "ISM-Tree is ignoring Z-Niveau for recognitionresult visualization.", True)
gen.add("useSceneColoring", bool_t, 0, "ISM-Tree is colored uniformly.", True)
gen.add("useZOffsetMax", bool_t, 0, "ISM-Tree is ignoring Z-Niveau but uses max function.", True)

exit(gen.generate(PACKAGE, "asr_ism_visualizations", "ism_result_visualizer"))
